Refactored OpenAPIUIBundlesUpdater to use service listeners for OpenA…

KeenanMcCarthy opened the pull request:
…PI-3.1

Modified OpenAPIUIBundlesUpdater in com.ibm.ws.microprofile.openapi.impl.OpenAPIUIBundlesUpdater

Commit:
Refactored microprofile openAPI and openapi.3.1 to use service listeners

Added OpenAPIUIBundlesListener to OpenAPIUIBundlesUpdater in microprofile openapi-3.1
Updated OpenAPI bundles updater to use service listners in openapi.3.1
Refactored OpenAPIUIBundlesUpdater to use OpenAPIUIBundleUpdater in com.ibm.ws.openapi.3.1
Added 2023 header to Bundles Listener

KeenanMcCarthy commented:
#libby
#build

tjwatson commented on dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesListener.java:
```diff
@@ -0,0 +1,81 @@
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
```
The header needs to be updated to use EPL-2.0, also use copyright date of 2023 now.

```suggestion
/*******************************************************************************
 * Copyright (c) 2023 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 * 
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
```

tjwatson reviewed (commented):
Looks like `com.ibm.ws.openapi31.customization.OpenAPIUIBundlesUpdater` also needs updated.

tjwatson commented on dev/com.ibm.ws.openapi.3.1/src/com/ibm/ws/openapi31/customization/OpenAPIUIBundlesListener.java:
```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
```
Need the EPL-2.0 header

tjwatson commented on dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesUpdater.java:
```diff
@@ -32,7 +32,6 @@
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.BundleException;
 import org.osgi.framework.FrameworkUtil;
-import org.osgi.framework.ServiceReference;
```
The header here looks to still be using EPL-1.0.  It looks like you need to rebase your PR branch onto the latest `release` branch to be sure you are basing on the latest license headers.

tjwatson replied:
Also, update the copyright date once you rebase for the files you modified.

Commit:
Updated copyright headers to EL-2.0, 2023

Updated copyright headers for OpenAPIUIBundles updater and OpenAPIUIBundlesListener in com.ibm.ws.openapi.3.1 and com.ibm.ws.microprofile.openapi

KeenanMcCarthy commented:
#libby

KeenanMcCarthy commented:
#build
